### PR TITLE
[libtar] new port

### PR DESCRIPTION
**Pull request by talregev**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libtar/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+vcpkg_configure_make(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_install_make()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include"  "${CURRENT_PACKAGES_DIR}/debug/share")
+
+file(INSTALL "${SOURCE_PATH}/COPYRIGHT" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (FrankXie05) [inline ports/libtar/portfile.cmake:16]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/COPYRIGHT")
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Review by FrankXie05 — DISMISSED** [review]

File: ports/libtar/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,16 @@
+set(VCPKG_USE_HEAD_VERSION ON)
```
Comment (vicroms) [inline ports/libtar/portfile.cmake:1]: why is this necessary?

Comment (talregev) [reply ports/libtar/portfile.cmake:1]: I think for this:     HEAD_REF v${VERSION}
But I can remove it.

**Comment by talregev** [comment]
@vicroms 
I add `set(VCPKG_USE_HEAD_VERSION ON)` because it fail to fetch it with REF var. 
I restore the line as before.

File: ports/libtar/portfile.cmake (hunk 2 above)
Comment (talregev) [reply ports/libtar/portfile.cmake:1]: I add set(VCPKG_USE_HEAD_VERSION ON) because it fail to fetch it with REF var.
I restore the line as before.

**Comment by dg0yt** [comment]
Can you at least try `REF 50dcdd27a33444b974d940ed5f225a2444e5aeba` (which is v1.2.20)?
This repo hasn't been updated in 9 years...

**Comment by FrankXie05** [comment]
See https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/maintainers/functions/vcpkg_from_git.md?plain=1#:~:text=vcpkg_from_git(,)

**Comment by talregev** [comment]
> Can you at least try `REF 50dcdd27a33444b974d940ed5f225a2444e5aeba` (which is v1.2.20)? This repo hasn't been updated in 9 years...

Yes, I guess I will do it. I prefer it with version.

**Comment by talregev** [comment]
@vicroms @FrankXie05 
Ready for review.

**Comment by talregev** [comment]
@FrankXie05 Please take a look. Don't let me wait for simple thing. I already change and remove that line.

**Comment by talregev** [comment]
@FrankXie05 Please remove the label `author-response`

**Review by FrankXie05 — DISMISSED** [review]

**Comment by FrankXie05** [comment]
Sorry about that. :)

File: ports/libtar/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,15 @@
+vcpkg_from_git(
+    OUT_SOURCE_PATH SOURCE_PATH
+    URL https://repo.or.cz/libtar.git
+    REF 50dcdd27a33444b974d940ed5f225a2444e5aeba #v1.2.20
```
Comment (vicroms) [inline ports/libtar/portfile.cmake:4]: There are a few memory leak fixes and invalid dereference fixes added since that release. We should use the latest commit in the repo to ensure we have those fixes in (you can keep the version number the same).

**Review by vicroms — CHANGES_REQUESTED** [review]

File: ports/libtar/portfile.cmake (hunk 3 above)
Comment (dg0yt) [reply ports/libtar/portfile.cmake:4]: > (you can keep the version number the same).

I would rather suggest a suffix like `+20131208`. Repology and others don't like lying about the version.

Comment (talregev) [reply ports/libtar/portfile.cmake:4]: > There are a few memory leak fixes and invalid dereference fixes added since that release. We should use the latest commit in the repo to ensure we have those fixes in (you can keep the version number the same).

How I can use the latest commit? 
Can I use github source? 
https://github.com/tklauser/libtar

Comment (talregev) [reply ports/libtar/portfile.cmake]: I updated to the latest commit.

**Comment by talregev** [comment]
@FrankXie05 @vicroms Can you review again?

**Review by vicroms — APPROVED** [review]

**Comment by dg0yt** [comment]
> > (you can keep the version number the same).

I repeat: Bad practice.
